src/main/java/com/google/step/servlets/UserServlet.java

```diff
@@ -151,4 +158,13 @@ private void updateTagsFollowedBy(long userId, String tagsString) {
 
     followManager.updateFollowedTagIds(userId, tagIds);
   }
+
+  private boolean validateUserName(String username) {
+    if (username.isEmpty()) {
+      return false;
+    }
+    Pattern p = Pattern.compile(USERNAME_PATTERN);
+    Matcher m = p.matcher(username);
```

Comment:
Generally a bad idea to name variables this way. Use `pattern` & `matcher` instead.